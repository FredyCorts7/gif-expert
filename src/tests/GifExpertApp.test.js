import React from 'react';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import '@testing-library/jest-dom';

describe('Testing to <GifExpertApp/>', () => {
  test('should be the same component', () => {
    const wrapper = shallow(<GifExpertApp />);

    expect(wrapper).toMatchSnapshot();
  });

  test('should show a list of categories', () => {
    const categories = ['One Piece', 'Kimetsu no Yaiba'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });
});
