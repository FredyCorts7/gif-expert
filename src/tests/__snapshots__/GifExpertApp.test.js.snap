// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <GifExpertApp/> should be the same component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button>
    Add category
  </button>
  <ol />
</Fragment>
`;

exports[`Testing to <GifExpertApp/> should show a list of categories 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <button>
    Add category
  </button>
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="Kimetsu no Yaiba"
      key="Kimetsu no Yaiba"
    />
  </ol>
</Fragment>
`;
